packages/jest-runtime/src/ScriptTransformer.js

```diff
@@ -208,8 +208,13 @@ class ScriptTransformer {
 
       if (typeof processed === 'string') {
         transformed.code = processed;
-      } else {
+      } else if (processed && typeof processed.code === 'string') {
```

Comment:
It'll fail for `undefined`. Unless we use `!= null`, but I haven't check if your eslint config allows `eqeq null`